### DISCUSSION TITLE
Get hostname from Active Directory DNS?

### DISCUSSION THREAD

**Discussion by codenotworking** [MichaCo/DnsClient.NET#224] [Q&A]
Is it possible for `DnsClient.NET` to get the hostname (A record) from Active Directory DNS (not LDAP)?

```csharp
// Basic C# code

using DnsClient;
using System.Net;

// AD DNS IP Address
LookupClient client = new(IPAddress.Parse("10.0.98.22"));

IDnsQueryResponse queryResponse = client.Query("some_computer_name", QueryType.A);

foreach (var record in queryResponse.Answers.ARecords()) {
    // This doesn't return anything
    Console.WriteLine(record.Address);
}
```

**Comment by MichaConrad** [comment]
Sure. As far as know, you still have to install and/or configure a DNS server based on you AD DS, see:
https://learn.microsoft.com/en-us/windows-server/networking/dns/quickstart-install-configure-dns-server?tabs=powershell

> When you install Active Directory Domain Services (AD DS) with the Active Directory Domain Services Installation Wizard, the wizard gives you the option to automatically install and configure a DNS server. The resulting DNS zone is integrated with the AD DS domain namespace. To learn more, see [Understanding Active Directory Domain Services Integration](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-r2-and-2008/cc726034(v=ws.11)).

But if you have that, then yeah, you should be able to query by using the IP (and Port if not default) of the DNS server

**Comment by codenotworking** [comment]
Yes, the DNS is configured for Active Directory.  I'm able to use the management console at work and I use `Get-DnsServerResourceRecord` PowerShell cmdlet.  But I'm trying to develop a C# app to do the same as the PowerShell cmdlet.

**Comment by MichaConrad** [comment]
To be clear, those powershell commands are of administrative purpose and allow you to query across zones and create or edit entries and what not. 
You'll not be able to do that via the DNS protocol (this library).

This library also does not extend the queries with the DNS servers default suffix, yet. So, to query for "some_computer_name", you'd have to append the dns suffix to that query to make it a valid dns query.
The NameServer object does have that as a property.

**Comment by codenotworking** [comment]
Thanks for the information.

> To be clear, those powershell commands are of administrative purpose and allow you to query across zones and create or edit entries and what not.

Do you have any recommendation as to how I can achieve the same results as `Get-DnsServerResourceRecord` PowerShell cmdlet in C#?
